src/odoc/indexing.ml

```diff
@@ -66,3 +73,58 @@ let compile ~output ~warnings_options inputs_in_file inputs =
   result |> Error.handle_warnings ~warnings_options >>= fun (_ : bool) ->
   Format.fprintf output "]";
   Ok ()
+
+let compile_to_marshall ~output ~warnings_options files =
+  let final_index = H.create 10 in
+  let unit u =
+    Odoc_model.Fold.unit
+      ~f:(fun () item ->
+        let entries = Odoc_search.Entry.entries_of_item item in
+        List.iter
+          (fun entry -> H.add final_index entry.Odoc_search.Entry.id entry)
+          entries)
+      () u
+  in
+  let page p =
+    Odoc_model.Fold.page
+      ~f:(fun () item ->
+        let entries = Odoc_search.Entry.entries_of_item item in
+        List.iter
+          (fun entry -> H.add final_index entry.Odoc_search.Entry.id entry)
+          entries)
+      () p
+  in
+  let index i = H.iter (H.add final_index) i in
+  let index () =
+    List.fold_left
+      (fun acc file ->
+        match handle_file ~unit ~page ~occ:index file with
+        | Ok acc -> acc
+        | Error (`Msg m) ->
+            Error.raise_warning ~non_fatal:true
+              (Error.filename_only "%s" m (Fs.File.to_string file));
+            acc)
+      () files
+  in
+  let result = Error.catch_warnings index in
+  result |> Error.handle_warnings ~warnings_options >>= fun () ->
+  Ok (Odoc_file.save_index output final_index)
+
+let compile out_format ~output ~warnings_options ~includes_rec ~inputs_in_file
+    ~odocls =
+  parse_input_files inputs_in_file >>= fun files ->
+  let files = List.rev_append odocls files in
+  let files =
+    List.rev_append files
+      (includes_rec
+      |> List.map (fun include_rec ->
+             Fs.Directory.fold_files_rec ~ext:"odocl"
+               (fun files file -> file :: files)
+               [] include_rec)
+      |> List.concat)
+  in
+  if files = [] then Error (`Msg "No .odocl files were included")
```

Comment:
As there's several indirect ways to pass the list of inputs, to reassure driver authors, we could log the number of odocl files considered for indexing:
```
Indexing N odocl files.
```